src/net/fe/unit/Unit.java

```diff
@@ -913,6 +913,17 @@ public void setHp(int hp) {
 			}
 		}
 	}
+	
+	/**
+	 * Sets the unit's max HP (used for Sudden Death)
+	 * 
+	 */
+	public void setMaxHp(int mhp){
+		this.stats.maxHp = mhp;
```

Comment:
What I'd do is, instead of assigning a new value to `this.stats.maxHP` and necessitate  `Statistics::maxHp` become modifiable, is assign a new value to `this.stats`, which is already modifiable.

`Statistics::copy` follows a Scala idiom not a Java one, I know, but it is used to get a new Statistics with one little thing changed (in this case, a new maxHp)